### ISSUE TITLE
Bug with launching camera/gallery

### ISSUE THREAD

**Issue by geeky-do** [bitstadium/HockeySDK-Android#330]
Hi, 

I just noticed an issue that was introduced to our application with the update from SDK 4.1.5 to version 5.0.2. I can confirm this bug was introduced with version 5. The issue seems to be that permission granting process was somehow altered in version 5 (I did some digging and I suspect this commit might be the trigger https://github.com/bitstadium/HockeySDK-Android/pull/292/commits/cd16f158d1ca794bdef99fec93747c8533e71050)

With version 4.1.5 I see the following logs related to launching the camera which is the correct flow we expect. 
`11-13 12:06:41.888 1692-2460/system_process I/ActivityManager: START u0 {act=android.media.action.IMAGE_CAPTURE flg=0x3 cmp=com.android.camera2/com.android.camera.CaptureActivity clip={text/uri-list U:content://media/external/images/media/97} (has extras)} from uid 10132 on display 0
11-13 12:06:42.091 1692-1716/system_process I/ActivityManager: Displayed com.android.camera2/com.android.camera.CaptureActivity: +189ms
`

However for version 5.x we see the following: 
`11-13 12:13:28.415 1692-2104/system_process I/ActivityManager: START u0 {act=android.content.pm.action.REQUEST_PERMISSIONS pkg=com.android.packageinstaller cmp=com.android.packageinstaller/.permission.ui.GrantPermissionsActivity (has extras)} from uid 10132 on display 0
`

FYI in both cases the permission was already granted and the app should be able to use it.

Please let me know if any additional logs would be useful for you.

**Comment by TroubleMakerBen** [comment]
Hey Dina,

Thanks for getting in touch.
I just saw your message here.
Apologies for the issue you're seeing.

HA 5.0.0 has a breaking change related to permissions which could cause the issue that you are seeing.
Can you add `<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" tools:node="replace"/>`to your manifest file as mentioned in the [release notes](https://github.com/bitstadium/HockeySDK-Android/releases/tag/5.0.0)?

Let me know if that helps – and please let us know if it doesn't help.

Benjamin

**Comment by M-Khay** [comment]
Yes This fix solves the problem, I had a similar issue and was resolved by this solution. Although, I see the issue being persisted in 5.0.3 as well. Can you provided any details as to why ?

**Comment by geeky-do** [comment]
Yes this works for us. Thanks for the help!

**Comment by MatkovIvan** [comment]
Hi @M-Khay,

We have an explanation in the [release notes](https://github.com/bitstadium/HockeySDK-Android/releases/tag/5.0.0):
> To be ready for Android O, HockeySDK-Android now limits the `WRITE_EXTERNAL_STORAGE` permission with the `maxSdkVersion` filter. In some use cases, e.g. where an app contains a dependency that requires this permission, `maxSdkVersion` makes it impossible for those dependencies to grant or request the permission. The solution for those cases is to declare the `tools:node="replace"` manifest merging strategy later in the dependency tree:
`<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" tools:node="replace"/>`

Let me know if you need more details about something specific, I'll happy to help!

BTW since this is not the first question about it, I wonder where you first look at the information about the changes in the new version? I'm interested in how to make this information easier to find in the future.
